src/zenml/integrations/mlflow/mlflow_step_decorator.py

```diff
@@ -0,0 +1,158 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+import functools
+from typing import Any, Callable, Optional, Type, TypeVar, Union, cast, overload
+
+from zenml.integrations.mlflow.mlflow_utils import (
+    get_or_create_mlflow_run,
+    setup_mlflow,
+)
+from zenml.logger import get_logger
+from zenml.steps import BaseStep, StepEnvironment
+from zenml.steps.utils import STEP_INNER_FUNC_NAME
+
+logger = get_logger(__name__)
+
+
+# step entrypoint type
+F = TypeVar("F", bound=Callable[..., Any])
+
+# step class type
+S = TypeVar("S", bound=Type[BaseStep])
+
+
+@overload
+def enable_mlflow(
+    _step: S,
+) -> S:
+    """Type annotations for mlflow step decorator in case of no arguments."""
```

Comment:
For the api docs it might be nice to include an Args: description - not sure if its necessary though

Reply:
Luckily, the API docs is smart enough to detect and ignore @override cases. We have a similar situation with `enable_whylogs` and the API docs are only rendered for the concrete function: https://apidocs.zenml.io/0.6.0/api_docs/integrations/#zenml.integrations.whylogs.whylogs_step_decorator

Reply:
Nice :)